doc/source/guide/derivatives_guide.rst

```diff
@@ -52,25 +57,31 @@ Here :math:`f'(x)` is defined as the linear operator (if it exists) that satisfi
 
 This definition of the derivative is called the *Fréchet derivative*.
 
+The Gateaux derivative
+~~~~~~~~~~~~~~~~~~~~~~
+The concept of directional derivative can also be extended to Banach spaces, giving the *Gateaux* derivative. 
+The Gateaux derivative is more general than the Fréchet derivative, but is not a linear operator. An example of a function that is Gateaux differentiable but not Fréchet differentiable is the absolute value function.
+It not used in ODL since it can fail to be linear. 
```

Comment:
Then this needs to be fixed:

https://github.com/odlgroup/odl/blob/b387ddb7d2eb104d4d801ac411082cde991a79fd/odl/diagnostics/operator.py#L302-L304

I thought we assumed the derivative to be linear across the board. Of course users can duck-type it, but thats the assumption we work with.